compliance_checker/acdd.py

```diff
@@ -634,6 +636,28 @@ def __init__(self):
             'publisher_institution',
         ])
 
+        # overrride the ISO date checks in
+        def _check_attr_is_iso_date(attr, ds):
+            result_name = "{}_is_iso".format(attr)
+            if not hasattr(ds, attr):
+                return ratable_result((0,2), result_name,
+                                      ["Attr {} is not present".format(attr)])
+            else:
+                iso_check, msgs = datetime_is_iso(getattr(ds, attr))
+                return ratable_result((1 + iso_check, 2),
+                                      result_name, msgs)
+
+        # run ISO 8601 date checks against the date_created, date_issued,
+        # date_modified, and date_metadata_modified global attributes
+        self.rec_atts = kvp_convert(self.rec_atts)
+        self.rec_atts['date_created'] = partial(_check_attr_is_iso_date,
+                                                'date_created')
+        self.sug_atts = kvp_convert(self.sug_atts)
+        for k in ("date_{}".format(suffix) for suffix in
+                  ('issued', 'modified', 'metadata_modified')):
```

Comment:
Maybe i'm old-school but this seems unnecessarily complex.

In my opinion, something like this is easier to follow:
```
date_attrs = ['date_issued', 'date_modified', 'date_metadata_modified']
for date_attr in date_attrs:
```